### ISSUE TITLE
media.currentTime doesn't reset to 0 when playlist ends

### ISSUE THREAD

**Issue by danielr18** [benwiley4000/cassette#330]
The currentTime in state is reset to 0, but the one in the media element isn't, would you accept a PR to fix this?

**Comment by benwiley4000** [comment]
Hm, I think I must have only been considering the case where the active track index goes back to 0 before stopping playback.

It seems to me like maybe the currentTime *shouldn't* reset to 0 if we're just staying on the last track? What do you think?

**Comment by benwiley4000** [comment]
@danielr18 I tried to repro this bug and I can't. Could you share a repro or at least tell me what browser you're running? Maybe I didn't understand what the issue was. Here are the scenarios I tested:

## Scenario A
Conditions:
- `loadFirstTrackOnPlaylistComplete`: `false`
- `state.cycle`: `false`
Outcome on last track ended:
Stays on last track, `state.currentTime` is *not* reset to 0, `media.currentTime` is *not* reset to 0

## Scenario B
Conditions:
- `loadFirstTrackOnPlaylistComplete`: `true`
- `state.cycle`: `false`
Outcome on last track ended:
Loads first track, `state.currentTime` *is* reset to 0, `media.currentTime` *is* reset to 0.

Browser: Chrome 70

**Comment by benwiley4000** [comment]
@danielr18 were you still interested in helping with this?

**Comment by danielr18** [comment]
Yes, your suggestion seems to work well for all cases, although it's a breaking change, you could call `onSelectTrackIndex` with the `activeTrackIndex` before and it wouldn't reload the media element, however 2.0.0 is still alpha, so these should be expected.

I'll have the PR created today.

**Comment by benwiley4000** [comment]
Hm.. that's a good point. I hadn't considered that. And I do want to preserve that behavior. If someone clicks on a track they're already listening to it shouldn't reload. We just want that to happen when the track has actually finished and it's restarting.

Here's an idea... maybe `getGoToTrackState` should take an extra boolean argument called `forceNewTrackLoad` which is `false` by default and only passed as `true` when it's called internally by backSkip or forwardSkip? And then `awaitingNewTrack` will be based on that.

Thanks for helping with this!

**Comment by danielr18** [comment]
Yes, makes sense, basically like `shouldPlay`.

**Comment by benwiley4000** [comment]
Yeah, right. Maybe it should also start with `should` for consistency.. `shouldForceTrackLoad`?

**Comment by danielr18** [comment]
or `shouldTrackReload`

**Comment by danielr18** [comment]
We might not have to include `track` in the variables we are introducing, since what it really does is reloading the media element (media.load), and it would be more consistent, we don't have `shouldTrackPlay`. 

Thus I think `shouldReload`/`shouldForceLoad` and `awaitingLoad` are better names.

**Comment by benwiley4000** [comment]
Released in v2.0.0-alpha.13.
